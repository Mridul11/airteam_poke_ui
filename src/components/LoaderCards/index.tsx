import { Skeleton, Card, Avatar } from 'antd';
import { useState } from 'react';

const { Meta } = Card;

const LoaderCards = () => {
  const [loader, _] = useState(true);

  return (
    <>
      <Card style={{ width: 300, marginTop: 16 }} loading={loader}>
        <Meta
          avatar={<Avatar src='https://joeschmoe.io/api/v1/random' />}
          title='Card title'
          description='This is the description'
        />
      </Card>

      <Card style={{ width: 300, marginTop: 16 }}>
        <Skeleton loading={loader} avatar active>
          <Meta
            avatar={<Avatar src='https://joeschmoe.io/api/v1/random' />}
            title='Card title'
            description='This is the description'
          />
        </Skeleton>
      </Card>
    </>
  );
};

export default LoaderCards;
