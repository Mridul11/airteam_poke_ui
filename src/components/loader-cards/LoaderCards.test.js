import { render, screen } from '@testing-library/react';
import LoaderCards from '.';
import { MemoryRouter } from 'react-router-dom';
import AddWatchMedia from '../../utils/test-utils/add-matchmedia';

describe('Loadercards component', () => {
  beforeAll(AddWatchMedia);
  test('renders loader-cards element', () => {
    render(
      <MemoryRouter>
        <LoaderCards />
      </MemoryRouter>
    );
    const loaderCardElement = screen.getByTestId('test-loadercard');
    expect(loaderCardElement).toMatchSnapshot();
  });
});
