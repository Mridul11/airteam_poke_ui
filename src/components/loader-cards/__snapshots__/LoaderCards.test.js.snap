// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loadercards component renders loader-cards element 1`] = `
<div
  data-testid="test-loadercard"
>
  <div
    class="ant-card ant-card-loading ant-card-bordered"
    style="width: 300px; margin-top: 16px;"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-loading-content"
      >
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-22"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-8"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-15"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-18"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-13"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-9"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-3"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-16"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-card ant-card-bordered"
    style="width: 300px; margin-top: 16px;"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
      >
        <div
          class="ant-skeleton-header"
        >
          <span
            class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
          />
        </div>
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width: 50%;"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
